src/services/sync/strategies/TSP/TSPSchoolSyncer.js

```diff
@@ -452,26 +526,28 @@ class TSPSchoolSyncer extends mix(Syncer).with(ClassImporter) {
 	 */
 	createOrUpdateClasses(classes, school, teacherMapping, classMapping) {
 		return Promise.all(
-			classes.map((klass) => {
-				const sourceOptions = {};
-				sourceOptions[SOURCE_ID_ATTRIBUTE] = klass.klasseId;
-				const query = {
-					source: ENTITY_SOURCE,
-					sourceOptions,
-				};
-				const teacher = teacherMapping[klass.lehrerUid];
-				const options = {
-					name: klass.klasseName,
-					schoolId: school._id,
-					year: school.currentYear,
-					teacherIds: teacher ? [teacher] : [],
-					userIds: classMapping[klass.klasseId] || [],
-					source: ENTITY_SOURCE,
-					sourceOptions,
-				};
-				const onlyAddNew = this.config.lastChange !== undefined;
-				return this.createOrUpdateClass(options, query, onlyAddNew); // see ClassImporter mixin
-			})
+			classes.map((klass) =>
+				limit(() => {
+					const sourceOptions = {};
```

Comment:
Can you oursource the body of the function in a private method with name that describe what should be executed?